h5py/_hl/files.py

```diff
@@ -286,21 +288,23 @@ def atomic(self, value):
             # pylint: disable=missing-docstring
             self.id.set_mpi_atomicity(value)
 
-    if swmr_support:
-        @property
-        def swmr_mode(self):
-            """ Controls single-writer multiple-reader mode """
-            return self._swmr_mode
+    @property
+    @with_phil
+    def swmr_mode(self):
+        """ Controls single-writer multiple-reader mode """
+        return swmr_support and bool(self.id.get_intent() & (h5f.ACC_SWMR_READ | h5f.ACC_SWMR_WRITE))
 
-        @swmr_mode.setter
-        @with_phil
-        def swmr_mode(self, value):
-            # pylint: disable=missing-docstring
+    @swmr_mode.setter
+    @with_phil
+    def swmr_mode(self, value):
+        # pylint: disable=missing-docstring
+        if swmr_support:
             if value:
                 self.id.start_swmr_write()
-                self._swmr_mode = True
             else:
                 raise ValueError("It is not possible to forcibly switch SWMR mode off.")
+        else:
+            raise RuntimeError('SWMR support is not available for version {}.{}.{}.'.format(*hdf5_version))
```

Comment:
```suggestion
            raise RuntimeError('SWMR support is not available in HDF5 version {}.{}.{}.'.format(*hdf5_version))
```

Just to make it totally clear that the version number refers to HDF5, not h5py.